dos/src/main.rs

```diff
@@ -284,34 +513,206 @@ pub mod test {
             timestamp(),
         )];
         let entrypoint_addr = nodes[0].gossip;
+
         run_dos(
             &nodes,
             1,
-            entrypoint_addr,
-            "random".to_string(),
-            10,
-            "tvu".to_string(),
             None,
+            DosClientParameters {
+                entrypoint_addr,
+                mode: Mode::Tvu,
+                data_size: 10,
+                data_type: DataType::Random,
+                data_input: None,
+                skip_gossip: false,
+                allow_private_addr: false,
+                transaction_params: TransactionParams::default(),
+            },
         );
 
         run_dos(
             &nodes,
             1,
-            entrypoint_addr,
-            "repair_highest".to_string(),
-            10,
-            "repair".to_string(),
             None,
+            DosClientParameters {
+                entrypoint_addr,
+                mode: Mode::Repair,
+                data_size: 10,
+                data_type: DataType::RepairHighest,
+                data_input: None,
+                skip_gossip: false,
+                allow_private_addr: false,
+                transaction_params: TransactionParams::default(),
+            },
         );
 
         run_dos(
             &nodes,
             1,
-            entrypoint_addr,
-            "repair_shred".to_string(),
-            10,
-            "serve_repair".to_string(),
             None,
+            DosClientParameters {
+                entrypoint_addr,
+                mode: Mode::ServeRepair,
+                data_size: 10,
+                data_type: DataType::RepairShred,
+                data_input: None,
+                skip_gossip: false,
+                allow_private_addr: false,
+                transaction_params: TransactionParams::default(),
+            },
+        );
+    }
+
+    #[test]
+    fn test_dos_local_cluster_transactions() {
+        let num_nodes = 1;
+        let cluster =
+            LocalCluster::new_with_equal_stakes(num_nodes, 100, 3, SocketAddrSpace::Unspecified);
+        assert_eq!(cluster.validators.len(), num_nodes);
+
+        let nodes = cluster.get_node_pubkeys();
+        let node = cluster.get_contact_info(&nodes[0]).unwrap().clone();
+        let nodes_slice = [node];
+
+        // send random transactions to TPU
+        // will be discarded on sigverify stage
+        run_dos(
+            &nodes_slice,
+            1,
+            None,
+            DosClientParameters {
+                entrypoint_addr: cluster.entry_point_info.gossip,
+                mode: Mode::Tpu,
+                data_size: 1024,
+                data_type: DataType::Random,
+                data_input: None,
+                skip_gossip: false,
+                allow_private_addr: false,
+                transaction_params: TransactionParams::default(),
+            },
+        );
+
+        // send transactions to TPU with varied number of random signatures
```

Comment:
"varied number of" confuses me here, and below, since I see that `num_sign` is set to a concrete number.

Reply:
Addressed